vitest.config.ts

```diff
@@ -0,0 +1,14 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+  test: {
+    environment: "happy-dom",
+    include: ["./src/tests/unit/*.test.(ts|tsx)"],
```

Comment:
Consider if we want to use the `tsx` suffix for test files.

I am not really sure whether they can/should be considered `tsx` even though we generally seem to use the `ts` and `tsx` suffixes without much consistency and it does not seem to make much difference functionally either.

Reply:
The convention is to use tsx when jsx syntax is present in the file.
We might need to test components in the future, so I think we should keep tsx for now.